components/suggest/src/db.rs

```diff
@@ -731,11 +754,16 @@ impl<'a> SuggestDao<'a> {
         &mut self,
         record_id: &SuggestRecordId,
         suggestions: &[DownloadedAmpWikipediaSuggestion],
+        fields: SuggestRecord,
     ) -> Result<()> {
         for suggestion in suggestions {
             self.scope.err_if_interrupted()?;
             let common_details = suggestion.common_details();
-            let provider = suggestion.provider();
+            let provider = if let SuggestRecord::AmpMobile = fields {
+                SuggestionProvider::AmpMobile
+            } else {
+                suggestion.provider()
+            };
```

Comment:
the clunky-ness continues here, since  `suggestion.providers()` differentiates amp and wikipedia by the advertiser (I think), and AMP mobile and AMP are the exact same in terms of structure?

Reply:
Yeah, I think that's why I'm leaning toward having a separate `insert_amp_mobile_suggestions` method. `insert_amp_wikipedia_suggestions` has to disentangle AMP Desktop and Wikipedia suggestions, and with the addition of AMP Mobile, this method now handles three different cases. A separate method would bring it down to two.